filter_plugins/filters.py

```diff
@@ -97,6 +97,19 @@ def _interface_check(context, interface, interface_type=None):
         elif fact_address:
             return _fail("Interface %s has an IPv4 address but none was "
                          "requested" % device)
+    
+    # Static IPv6 address
+    if interface.get("bootproto") == "static" and interface.get("ip6"):
+        fact_address = fact.get("ipv6", [])
+        # IP address
+        if len(fact_address) == 0:
+            return _fail("Interface %s has no IPv6 address" % device)
+        
+        for item in fact_address:
+            if item["address"] == interface["ip6"]["address"] and item["prefix"] == str(interface["ip6"]["prefix"]):
+                break
+        else:
+            return _fail("Interface %s has incorrect IPv6 address" % device)
```

Comment:
Should we also be checking the IPv6 gateway here, if one is set?

Reply:
I was a little confused by this one, and wanted to check with you eitherway. Ansible doesn't return any routing table information. If the gateway for a traffic interface is going to be different from the default gateway, is there any point in checking?

The same actually applies for the existing v4 gateway check as well. Why do it say in the case of traffic interface having a different network than mgmt interface?

Reply:
The `gateway` field is only for the default route - other routes should be provided via the `route` field. This applies to v4 and v6. For v4 we check against the `ansible_default_ipv4` fact. There does appear to be an `ansible_default_ipv6` fact, could we use that?

Reply:
I see what you are saying, and that makes sense. Yeah, that fact is available, and I will add the check.

Reply:
Actually, I do want to clarify on this gateway aspect. I understand the concept of a default gateway for a system as a whole. But gateways can be specified per interface as well right? 

The way the role is currently setup, routing info on any interface other than the mgmt interface needs to be setup using explicit routes. Nothing is stopping people from adding a gateway field per interface with a different value, but it will cause the ether_check to fail.

Is my understanding right?

Reply:
Right - it is currently possible to specify a gateway for more than one interface, and if you do then the check will fail.

It is possible in Linux to have more than one default route. But in practice this does not make much sense - how could two routes be the default? Routing is performed using the destination IP, not the interface.

One way to support multiple default routes is to use multiple route tables. I have set this up in the past, and in fact added support to this role to configure it. It's an unusual case though, and wouldn't recommend it unless you need it.